setup.py

```diff
@@ -142,7 +142,6 @@ def _set_builtin(name, value):
     [udunits_ext] = cythonize(
         udunits_ext,
         compiler_directives=COMPILER_DIRECTIVES,
-        language_level="3str",
```

Comment:
> @trexfeathers I think this may possibly be all that is needed ?
> 
> So far, this adopts exactly what Bill proposed :
> 
> * insist on Cython>=3
> * remove the "language_level" setting.
> 
> But .. I'm still not sure it wouldn't be better to simply _unpin_ Cython, and leave the language-level explicit (which is then not superfluous).
> 
> Particularly as I see that the build does raises a warning about the default having changed recently. `/Users/runner/work/cf-units/cf-units/.tox/py311-osx-test/lib/python3.11/site-packages/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /Users/runner/work/cf-units/cf-units/cf_units/_udunits2.pxd`
> 
> What do you think ?

Since one of the other suggestions in #360 was specifically focussed on suppressing a warning (the deprecated NumPy one), I'd say that it's equally worth including `language_level` here, **and a comment explaining** - it might seem redundant to someone coming in cold later.

Reply:
See https://github.com/SciTools/cf-units/pull/460/commits/a53ae2012cf3c98e644141bad5dcd05235d481c7